sigstore-java/src/main/java/dev/sigstore/timestamp/client/TimestampRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.timestamp.client;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import org.immutables.value.Value.Immutable;
+
+@Immutable
+public interface TimestampRequest {
+  /** The hash algorithm used to hash the artifact. */
+  String getHashAlgorithm();
+
+  /** The hash of the artifact to be timestamped. */
+  byte[] getHash();
+
+  /** A nonce to prevent replay attacks. */
+  BigInteger getNonce();
```

Comment:
we might want to just generate this by default, and not require the user to provide it?